beamer/agent/events.py

```diff
@@ -298,7 +305,7 @@ def _fetch_range(
     def fetch(self) -> list[Event]:
         try:
             block_data = self._web3.eth.get_block("latest")
-            block_number = BlockNumber(block_data["number"])
+            block_number = BlockNumber(block_data["number"] - self._confirmation_blocks)
```

Comment:
Nitpick: this is totally fine, but I'd prefer to have an `assert block_number > 0` after this anyway...
Also, only the `get_block` call should be inside of this `try...except` block, the rest does not need to be there so let's move it out of the block.

Reply:
not sure about assert here actually, lets say confirmation blocks 1 and block number is 1 then this makes 0 and assert will crash, i would have it as if and return [] instead so this wont crash

Reply:
not adding any assert would be doing it actually

Reply:
> not adding any assert would be doing it actually

I'm afraid I don't understand this. :sweat_smile:

Reply:
so block_number < 0 will return [] in next if at line 317 because self._next_block_number would be bigger

Reply:
> so block_number < 0 will return [] in next if at line 317 because self._next_block_number would be bigger

Ah, yes, perfect! :+1: